pithos/plugins/notify.py

```diff
@@ -45,6 +45,23 @@ def on_prepare(self):
         if 'action-icons' in caps:
             self.notification.set_hint('action-icons', GLib.Variant('b', True))
 
+        # GNOME Shell 3.20 or higher has bultin MPRIS functionality that makes
+        # persistent song notifications redundant.
+        has_built_in_mpris = False
+        if server_info['name'] == 'gnome-shell':
+            version = server_info['version'].split('.')
+            if version and len(version) >= 2:
+                try:
+                    major_version = int(version[0])
+                    minor_version = int(version[1])
+                except ValueError:
+                    major_version = minor_version = 0
+                if major_version >= 3 and minor_version >= 20:
```

Comment:
4.0 would represent a total change in API/ABI in GNOME versioning,  much like 2.x to 3.0. More than likely the whole app would need to be ported to 4.0.